eZ/Publish/Core/Search/Common/Slot/SwapLocation.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common\Slot;
+
+use eZ\Publish\Core\SignalSlot\Signal;
+
+/**
+ * A Search Engine slot handling SwapLocationSignal.
+ */
+class SwapLocation extends AbstractSubtree
+{
+    /**
+     * Receive the given $signal and react on it.
+     *
+     * @param Signal $signal
+     */
+    public function receive(Signal $signal)
+    {
+        if (!$signal instanceof Signal\LocationService\SwapLocationSignal) {
+            return;
+        }
+
+        $this->indexSubtree($signal->location1Id);
+        $this->indexSubtree($signal->location2Id);
```

Comment:
Do we need to re index the whole subtree? It's just the content object being swapped, not the tree.